modules/seqexec/server/src/main/scala/seqexec/server/SeqexecEngine.scala

```diff
@@ -461,7 +461,13 @@ object SeqexecEngine {
   }
 
   def giapiConnection: Kleisli[IO, Config, Giapi[IO]] = Kleisli { cfg: Config =>
-    Giapi.giapiConnectionIO.connect
+    val instSim = cfg.require[Boolean]("seqexec-engine.instSim")
+    val gpiUrl  = cfg.require[String]("seqexec-engine.gpiUrl")
+    if (instSim) {
+      Giapi.giapiConnectionIO.connect
```

Comment:
We could in principle use an `Id` based simulator as however this means generalizing most of the seqexec-server code from `IO` to `F[_]`. I started that fix but it is a fairly big change